test/const_eval_test.cpp

```diff
@@ -47,6 +47,10 @@ struct non_computable_cf
 struct test_context
 {
     void finish() const {}
+    void set_stream(int) {}
+    void create_events(int) {}
+    void record_event(int) {}
+    void wait_event(int) {}
```

Comment:
These method should be optional. So we dont have to add them everywhere.

Reply:
I originally did not add these to this test, but it fails somewhere during the call chain.

Reply:
That is because you added them to the `context` interface so the methods are required for every context created. Ideally, these methods should only be in the gpu context.